src/app/fdctl/run/tiles/fd_store_int.c

```diff
@@ -0,0 +1,575 @@
+  fd_mcache_seq_update( ctx->repair_req_out_sync, ctx->repair_req_out_seq );
+
+  ctx->store->now = fd_log_wallclock();
+
+  for( ulong i = fd_pending_slots_iter_init( ctx->store->pending_slots );
+         (i = fd_pending_slots_iter_next( ctx->store->pending_slots, ctx->store->now, i )) != ULONG_MAX; ) {
+    uchar const * block = NULL;
+    ulong         block_sz = 0;
+    ulong repair_slot = FD_SLOT_NULL;
+    int store_slot_prepare_mode = fd_store_slot_prepare( ctx->store, i, &repair_slot, &block, &block_sz );
+
+    ulong slot = repair_slot == 0 ? i : repair_slot;
+    FD_LOG_DEBUG(( "store slot - mode: %d, slot: %lu, repair_slot: %lu", store_slot_prepare_mode, i, repair_slot ));
+    fd_store_tile_slot_prepare( ctx, store_slot_prepare_mode, slot );
+  }
+}
+
+void
+unprivileged_init( fd_topo_t *      topo,
+                   fd_topo_tile_t * tile,
+                   void *           scratch ) {
+  if( FD_UNLIKELY( tile->in_cnt != 3 ||
+                   strcmp( topo->links[ tile->in_link_id[ SHRED_IN_IDX     ] ].name, "shred_storei" )    ||
+                   strcmp( topo->links[ tile->in_link_id[ REPAIR_IN_IDX ] ].name, "repair_store" ) ) )
+    FD_LOG_ERR(( "store tile has none or unexpected input links %lu %s %s",
+                 tile->in_cnt, topo->links[ tile->in_link_id[ 0 ] ].name, topo->links[ tile->in_link_id[ 1 ] ].name ));
+
+  if( FD_UNLIKELY( tile->out_cnt != 2 ||
+                   strcmp( topo->links[ tile->out_link_id[ REPAIR_OUT_IDX ] ].name, "store_repair" ) ||
+                   strcmp( topo->links[ tile->out_link_id[ REPLAY_OUT_IDX ] ].name, "store_replay" ) ) )
+    FD_LOG_ERR(( "store tile has none or unexpected output links %lu %s %s",
+                 tile->out_cnt, topo->links[ tile->out_link_id[ 0 ] ].name, topo->links[ tile->out_link_id[ 1 ] ].name ));
+      
+  if( FD_UNLIKELY( tile->out_link_id_primary != ULONG_MAX ) )
+    FD_LOG_ERR(( "store tile has a primary output link" ));
+
+  /* Scratch mem setup */
+
+  FD_SCRATCH_ALLOC_INIT( l, scratch );
+  fd_store_tile_ctx_t * ctx = FD_SCRATCH_ALLOC_APPEND( l, alignof(fd_store_tile_ctx_t), sizeof(fd_store_tile_ctx_t) );
+  // TODO: set the lo_mark_slot to the actual snapshot slot!
+  ctx->store = fd_store_join( fd_store_new( FD_SCRATCH_ALLOC_APPEND( l, fd_store_align(), fd_store_footprint() ), 1 ) );
+  ctx->repair_req_buffer = FD_SCRATCH_ALLOC_APPEND( l, alignof(fd_repair_request_t), MAX_REPAIR_REQS * sizeof(fd_repair_request_t) );
+  ctx->stake_ci = fd_stake_ci_join( fd_stake_ci_new( FD_SCRATCH_ALLOC_APPEND( l, fd_stake_ci_align(), fd_stake_ci_footprint() ), ctx->identity_key ) );
+  void * smem = FD_SCRATCH_ALLOC_APPEND( l, fd_scratch_smem_align(), fd_scratch_smem_footprint( SCRATCH_SMAX ) );
+  void * fmem = FD_SCRATCH_ALLOC_APPEND( l, fd_scratch_fmem_align(), fd_scratch_fmem_footprint( SCRATCH_SDEPTH ) );
+  ulong scratch_top = FD_SCRATCH_ALLOC_FINI( l, 1UL );
+  if( FD_UNLIKELY( scratch_top > (ulong)scratch + scratch_footprint( tile ) ) ) {
```

Comment:
same comments as other tiles, shift check to bottom and re-org init

Reply:
Done